README.md

```diff
@@ -1,2 +1,29 @@
-# fastcode
+# fastcode  [![Build Status](https://travis-ci.com/matthaeusheer/fastcode.svg?branch=master)](https://travis-ci.com/matthaeusheer/fastcode)
+
 Group project for ETH course "How to Write Fast Numerical Code".
+
+## Testing
+
+This project uses [`criterion`](https://github.com/Snaipe/Criterion) for testing. Unit
+tests should be written straight in the `main.c` file of the algorithm. Integration tests
+such as numerical validations can be written in the `tests/integration.c` file. In order to
+run all tests (unit and integration), please `make` all algorithms, then use:
+
```

Comment:
I think our unit tests should be separated into the tests folder following the same folder structure and should not live in the source files.  
So for **pengu/pengu.c** would have the tests live in **tests/pengu/test_pengu.c**.  
I saw that this applies to the feature/pengu_base branch as well.

Reply:
Personally I really like to have unit tests attached to the code they are testing. I guess we can talk about this with the others.

Reply:
Plus it makes test compilation much easier as it does not require an additional compilation command in the Makefile.

Reply:
Plus it allows to test `static` functions that are only visible in the implementation file. This allows to only expose the actual algorithm function and still test those "private" functions.

Reply:
Aight fair enough then let's keep it that way.